src/components/datagrid/data_grid_header_cell.tsx

```diff
@@ -311,70 +312,72 @@ export const OuiDataGridHeaderCell: FunctionComponent<OuiDataGridHeaderCellProps
   ) : null;
 
   return (
-    <div
-      role="columnheader"
-      {...ariaProps}
-      ref={headerRef}
-      tabIndex={isFocused && !isCellEntered ? 0 : -1}
-      className={classes}
-      data-test-subj={`dataGridHeaderCell-${id}`}
-      style={width != null ? { width: `${width}px` } : {}}>
-      {column.isResizable !== false && width != null ? (
-        <OuiDataGridColumnResizer
-          columnId={id}
-          columnWidth={width}
-          setColumnWidth={setColumnWidth}
-        />
-      ) : null}
+    <OuiToolTip position="top" content={id}>
```

Comment:
I feel like a `InnerText` would be a better method to get the contents rather than just using `id`. Maybe we could otherwise use `display || displayAsText || id` if using 2 refs is too much of an issue